include/multipass/exceptions/invalid_memory_size_exception.h

```diff
@@ -29,8 +29,8 @@ class InvalidMemorySizeException : public std::runtime_error
 {
 public:
     InvalidMemorySizeException(const std::string& val)
-        : runtime_error(fmt::format("{} is not a valid memory size - need a non-negative integer (in base 10) "
-                                    "followed by the desired unit (e.g. 1234B, 42MiB)",
+        : runtime_error(fmt::format("{} is not a valid memory size - need a non-negative decimal (in base 10) "
```

Comment:
And I think this needs rephrasing, otherwise it can get confusing:

```plain
$ multipass set local.teaching-sculpin.memory=1.1B
set failed: Invalid setting 'local.teaching-sculpin.memory=1.1B': 1.1B is not a valid memory size - need a non-negative decimal (in base 10) followed by the desired unit (e.g. 1234B, 42MiB, 0.5G)
```

Reply:
I am thinking perhaps: "need a non-negative integer, possibly followed by 'B', or a decimal followed by a larger unit (e.g. 1234B, 42MiB, 0.5G)"